react/features/face-tracking/components/FaceTracker.js

```diff
@@ -0,0 +1,267 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import tracking from 'tracking';
+
+import FacePrompt from './FacePrompt';
+
+const ANOMALY_RECOGNIZED_DELAY = 6000;
+const FACE_PROMPT_DURATION = 4000;
+
+const FACE_RECT_WIDTH_RATIO = 0.7;
+const FACE_RECT_HEIGHT_RATIO = 0.9;
+const FACE_RECT_TOP_RATIO = (1 - FACE_RECT_HEIGHT_RATIO) / 2;
+const FACE_RECT_LEFT_RATIO = (1 - FACE_RECT_WIDTH_RATIO) / 2;
+
+const TRACKING_INITIAL_SCALE = 4;
+const TRACKING_STEP_SIZE = 2;
+const TRACKING_EDGES_DENSITY = 0.1;
+
+/**
+ * React component for tracking face in a video, and calling FacePrompt
+ * component to display prompt if user's face is not well-positioned for a
+ * while.
+ *
+ * @extends Component
+ */
+class FaceTracker extends Component {
```

Comment:
The FaceTracker looks more like a middleware to me, not a component. You can have an action START_FACE_DETECTION, which is then caught by the face detection feature middleware and does it's magic and then dispatches actions to tell the FacePrompt to show itself or hide itself. There is no reason for FaceTracker to be a component.